rclcpp/src/rclcpp/signal_handler.cpp

```diff
@@ -113,8 +113,8 @@ SignalHandler::get_logger()
 SignalHandler &
 SignalHandler::get_global_signal_handler()
 {
-  static SignalHandler signal_handler;
-  return signal_handler;
+  static SignalHandler * signal_handler = new SignalHandler();
```

Comment:
I am not sure if this is the same behavior. this creates the new `SignalHandler` object every time it calls `SignalHandler::get_global_signal_handler()`. originally this was Singleton design, so it expects to return the same object within this class.

probably we need to do,

- having `static SignalHandler * signal_handler` in private member variable with `SignalHandler`.

and,

```suggestion
  if (signal_handler == NULL) {
    signal_handler = new SignalHandler();
    return signal_handler;
  } else {
    return signal_handler;
  }
```

Reply:
I think the `static ...` statement will only ever be run once, e.g. `static int i = 0;` will only result in `i` being set once.

That being said, I'm not sure this is the right approach. I'll need to understand the root issue better.

Reply:
Yeah, that was my first worry too. However, I double checked this with a simple /.a.out program. It seemed safe from multiple allocations.

Reply:
> I think the static ... statement will only ever be run once

I see, and that makes sense. thank you very much for the explanation.